src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -500,28 +500,35 @@ bool Snippet::created() const {
 }
 
 void Snippet::generate(const jit_snippets_compile_args* jcp) {
-    ov::pass::Manager optManager;
-    optManager.register_pass<ov::intel_cpu::pass::FuseLoadConvert>();
-    optManager.register_pass<ov::intel_cpu::pass::FuseStoreConvert>();
-    optManager.register_pass<ConvertToSwishCPU>();
-    optManager.register_pass<ov::intel_cpu::pass::MulAddToFMA>();
+    ov::pass::Manager pre_dialect;
+    pre_dialect.register_pass<ConvertToSwishCPU>();
 
+    ov::pass::Manager post_dialect;
+
+    ov::pass::Manager post_precision;
+    post_precision.register_pass<ov::intel_cpu::pass::FuseLoadConvert>();
+    post_precision.register_pass<ov::intel_cpu::pass::FuseStoreConvert>();
     // LoadConvert uses Load emitter that support conversion from any type to only f32
-    optManager.get_pass_config()->set_callback<ov::intel_cpu::pass::FuseLoadConvert>(
+    post_precision.get_pass_config()->set_callback<ov::intel_cpu::pass::FuseLoadConvert>(
             [](const std::shared_ptr<const ov::Node>& n) -> bool {
                 if (const auto& convert = std::dynamic_pointer_cast<const ov::op::v0::Convert>(n))
                     return convert->get_destination_type() != ov::element::f32;
                 return true;
             });
-
     // StoreConvert uses Store emitter that support conversion from only f32 to any types
-    optManager.get_pass_config()->set_callback<ov::intel_cpu::pass::FuseStoreConvert>(
+    post_precision.get_pass_config()->set_callback<ov::intel_cpu::pass::FuseStoreConvert>(
             [](const std::shared_ptr<const ov::Node>& n) -> bool {
                 if (const auto& convert = std::dynamic_pointer_cast<const ov::op::v0::Convert>(n))
                     return convert->get_input_element_type(0) != ov::element::f32;
                 return true;
             });
-    schedule = snippet->generate(optManager, reinterpret_cast<const void*>(jcp));
+    post_precision.register_pass<ov::intel_cpu::pass::MulAddToFMA>();
```

Comment:
Maybe we should register this pass in `post_dialect`? Because `FMA` emitter depends on precision as well. I understand that the emitter supports the same precisions as `Add` and `Mul` emitters but just in a case.

Reply:
thanks, have be discussed with @v-Golubev before, it's the best place for `MulAddToFMA`. I will add comments later.